scripts/prepare-crds

```diff
@@ -115,4 +115,165 @@ done
 # 	{{- required "Required CRDs are missing. Please install the ${name}-crd chart before installing this chart." "" -}}
 # {{- end -}}
 # {{- end -}}
-EOF
\ No newline at end of file
+EOF
+
+# Restructure the package to use Jobs to manage the CRD chart lifecyle instead of the templates/ directory if
+# the assumeOwnershipOfCRDs flag is enabled
+assume_ownership=$(cat ${f}/package.yaml | yq r - generateCRDChart.assumeOwnershipOfCRDs)
+if [[ "${assume_ownership}" != "true" ]]; then
+	exit 0
+fi
+
+mv ${f}/charts-crd/templates ${f}/charts-crd/crd-manifest
+mkdir -p ${f}/charts-crd/templates
+
+cat << EOF > ${f}/charts-crd/values.yaml
+# Default values for ${name}-crd.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+global:
+  systemDefaultRegistry: ""
+
+image:
+  repository: rancher/hyperkube
+  tag: v1.18.6-rancher1
+EOF
+
+cat << EOF > ${f}/charts-crd/templates/_helpers.tpl
+# Rancher
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- end -}}
+{{- end -}}
+EOF
+
+cat << EOF > ${f}/charts-crd/templates/rbac.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Chart.Name }}-manager
+  labels:
+    app: {{ .Chart.Name }}-manager
+rules:
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs: ['create', 'get', 'patch', 'delete']
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Chart.Name }}-manager
+  labels:
+    app: {{ .Chart.Name }}-manager
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ .Chart.Name }}-manager
+subjects:
+- kind: ServiceAccount
+  name: {{ .Chart.Name }}-manager
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Chart.Name }}-manager
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}-manager
+EOF
+
+cat << EOF > ${f}/charts-crd/templates/manifest.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Chart.Name }}-manifest
+data:
+  crd-manifest.yaml: |
+    {{- \$currentScope := . -}}
+    {{- \$crds := (.Files.Glob  "crd-manifest/**.yaml") -}}
+    {{- range \$path, \$_ :=  \$crds -}}
+    {{- with \$currentScope -}}
+    {{ .Files.Get \$path | nindent 4 }}
+    ---
+    {{- end -}}{{- end -}}
```

Comment:
I thought a ConfigMap was fine for storing the CRD manifest since there's no sensitive data here; does anyone feel strongly about making this a Secret instead?